sdk/azidentity/managed_identity_client.go

```diff
@@ -296,6 +306,29 @@ func (c *managedIdentityClient) createAppServiceAuthRequest(ctx context.Context,
 	return request, nil
 }
 
+func (c *managedIdentityClient) createAzureMLAuthRequest(ctx context.Context, id ManagedIDKind, scopes []string) (*policy.Request, error) {
+	request, err := runtime.NewRequest(ctx, http.MethodGet, c.endpoint)
+	if err != nil {
+		return nil, err
+	}
+	request.Raw().Header.Set("secret", os.Getenv(msiSecret))
+	q := request.Raw().URL.Query()
+	q.Add("api-version", "2017-09-01")
+	q.Add("resource", strings.Join(scopes, " "))
+	q.Add("clientid", os.Getenv(defaultIdentityClientID))
+	if id != nil {
+		if id.idKind() == miResourceID {
+			log.Write(EventAuthentication, "WARNING: Azure ML doesn't support specifying a managed identity by resource ID")
```

Comment:
Does this effectively become a no-op, or should this be a hard error?

Reply:
Personally I think it should be a hard error because we have no reason to believe we can realize the caller's intent and the result depends on how the server handles the undocumented and probably unexpected parameter. But I lost that argument and we took this approach across all languages, so I'm following along here for the sake of consistency.

Reply:
I'm hesitant to document this because the blurb would be complex (there are 3 other platforms on which we similarly set unspecified parameters), I believe it would generally be more distracting than helpful, and it could be rendered incorrect at any time by a service change we wouldn't observe. But if you think it's for the best, I'll take a stab at it.

Reply:
Good point. I suppose we can leave it as-is for now and reassess based on customer feedback.